graylog2-server/src/main/java/org/graylog2/plugin/indexer/retention/RetentionStrategyConfig.java

```diff
@@ -29,4 +30,8 @@ public interface RetentionStrategyConfig {
 
     @JsonProperty(MAX_NUMBER_OF_INDEXES_FIELD)
     int maxNumberOfIndices();
+
+    default void validate(ElasticsearchConfiguration elasticsearchConfiguration) throws IllegalArgumentException {
```

Comment:
I think valditation methods should not be implemented by throwing an exception.
Better return something like 
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/plugin/rest/ValidationResult.java